### ISSUE TITLE
VimR unexpectedly requesting access to contacts

### ISSUE THREAD

**Issue by adamneilson** [qvacua/vimr#691]
I'm not entirely sure why this is happening but since upgrading to MacOS Mojave (10.14), VimR is requesting access to my contacts. I can't think of a legitimate need to read my contacts?

Version: 0.24.0 (282)

![](http://s3.amazonaws.com/screenshots.aan.io/screenshots/Monosnap_2018-10-13_11-56-59.png)

**Comment by williamleuschner** [comment]
This also happened to me. What the heck? Was it doing this before too‽

**Comment by ActionScripted** [comment]
Similar, but for me it's asking for calendar permissions.

**Comment by numbcoder** [comment]
This also happened to me when vimr start from my home dir

**Comment by arpitchauhan** [comment]
This most likely has to do with the new privacy protections in macOS Mojave https://www.howtogeek.com/361707/how-macos-mojaves-privacy-protection-works/

**Comment by adamneilson** [comment]
Absolutely due to the new privacy protections but the point is **why** would VimR be attempting to access my contacts to begin with?

**Comment by ELLIOTTCABLE** [comment]
@adamneilson I'm sure it's just recursively listing your home-directory or something in some odd corner case.

Try running Ack or RipGrep for some common term starting at your home directory. You'll see something very similar, very quickly.

<img width="582" alt="image" src="https://user-images.githubusercontent.com/200/47836382-13a10800-dd76-11e8-83f9-124d8a297703.png">

**Comment by arpitchauhan** [comment]
Yeah, I have experience that too @ELLIOTTCABLE while using Terminal. And, I also vaguely remember other applications unexpectedly asking such permissions.

I think it'd be good if there could be a fix which could stop VimR from accessing photos, contacts etc, but I also think **if** your innocuous explanation is right (which I think it is), this is no big deal, rather just a minor annoyance.

**Comment by qvacua** [comment]
I updated to mojave (10.14.1) from high sierra today and I don't see any of that warnings. However, it may well be that, since I did not care for macOS entitlements and such, mojave thinks that VimR somehow wants to access everything... I'll have look into macOS security system to "fix" this. It could take awhile since I currently don't have much free time...

**Comment by adamdicarlo** [comment]
Since no one else has mentioned _sleep_: I've seen this a few times and I think it's only been when waking my (Mojave) system. "VimR.app would like to access your contacts" dialog box is waiting for me.

**Comment by xentek** [comment]
Don't start vimr in your home folder -- any app that crawls your directories from home will eventually hit `~/Library` and that's where the warnings are coming from, AFAIK. Like @ELLIOTTCABLE said, you can trigger this with ripgrep, etc and is not a an issue caused by VimR, but the user.

**Comment by williamleuschner** [comment]
@Xentek Okay, if this is my fault, how do I make it so that typing “vimr” into Spotlight and hitting enter launches it someplace other than `~`?
